File: problemsets/Codeforces/Python/A267.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
for s in [*open(0)][1:]:
  a,b=map(int,s.split())
  r=0
  while a and b:
    if b>a:r+=0--(b-a)//a;b%=a
    else:r+=0--(a-b)//b;a%=b
  print(r+1)

File: problemsets/Codeforces/C++/A267.cpp
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
#include <bits/stdc++.h>

using namespace std;

int main() {
  ios::sync_with_stdio(0);
  int n; cin >> n;
  while (n--) {
    int a, b; cin >> a >> b;
    int ret = 0;
    while (a && b) {
      if (a > b) ret += a/b, a %=b;
      else ret += b/a, b%=a;
    }
    cout << ret << endl;
  }
} 
